### ISSUE TITLE
[MQTT] Avoid mqtt retained reprocessing

### ISSUE THREAD

**Issue by logistark** [akka/alpakka#1767]
Hi, i am developing a service where i have to process Mqtt messages with retained flag. But i noticed that the messages with retained flag keep being reprocessed. I have been reading the logs of Mosquitto, and i see that when Alpakka Mqtt read receive a message it gets disconnected from it, and looks like it reconnects to Mosquitto and read again the retained message.

Is there any option or way to only get the retained messages once?

**Comment by huntc** [comment]
What QoS level are you using? If you read a retained message with QoS 1, but don’t get to ack, then you’ll keep receiving it.

**Comment by logistark** [comment]
I am using QoS 0 both at ```MqttSubscriptions.create(topic, MqttQoS.atMostOnce())``` and ```MqttSource.atMostOnce(settings, subs, size)```

**Comment by huntc** [comment]
Actually, the last message published to a topic will always be retained. So, if you keep subscribing, you’ll keep getting it. I’m sure that’s the correct behaviour.

**Comment by logistark** [comment]
Yes, that is the correct behaviour. But i don't know how to make Alpakka Mqtt to keep the subscription and connection to the broker in order to avoid reprocess the retained messages.

**Comment by huntc** [comment]
Are we talking about the mqtt-streaming project or just the mqtt one? The former factors out connection management whereas the latter only does it once a connection is established.

**Comment by logistark** [comment]
Mqtt one, but i want to subscribe only once and wait for messages. In the doc example with restart source. It subscribes every time it receives a message.
